RecoTracker/CkfPattern/src/CkfTrajectoryBuilder.cc

```diff
@@ -235,8 +139,19 @@ unsigned int CkfTrajectoryBuilder::limitedCandidates(const std::shared_ptr<const
           updateTrajectory(newTraj, std::move(*itm));
 
           if (toBeContinued(newTraj)) {
-            newCand.push_back(std::move(newTraj));
-            std::push_heap(newCand.begin(), newCand.end(), trajCandLess);
+            if (full) {
```

Comment:
Here now we immediately check if new candiate is better of the worst in newCand.
if yes, we swap, if not we do not push it.
Of course we keep the "heap" properly ordered.

In principle the result should be identical to the previous one, unless some candidates have identical score as the order in not the same anymore...

I remind that this TrajectoryBuilder is used only by electron and muons....